diff --git a/node_modules/mapbox-gl-inspect/lib/MapboxInspect.js b/node_modules/mapbox-gl-inspect/lib/MapboxInspect.js
index ea9bf07..3e7ab38 100644
--- a/node_modules/mapbox-gl-inspect/lib/MapboxInspect.js
+++ b/node_modules/mapbox-gl-inspect/lib/MapboxInspect.js
@@ -159,7 +159,12 @@ MapboxInspect.prototype._onSourceChange = function () {
   });
 
   if (!isEqual(previousSources, sources) && Object.keys(sources).length > 0) {
-    this.render();
+    setTimeout(() => {
+      // The calling method is being called when the sources are being updated.
+      // render will might call setStyle, which is not a good idea while updating the source.
+      // So, we are delaying the render to the next event loop.
+      this.render();
+    }, 100);
   }
 };
 
